rust/lance-encoding/src/testing.rs

```diff
@@ -131,31 +163,57 @@ pub async fn check_round_trip_field_encoding(mut encoder: impl FieldEncoder, fie
         let schema = Schema::new(vec![field.clone()]);
 
+        // Test range scheduling
         for range in [0..500, 100..1100, 8000..8500] {
+            let range = range.start as u64..range.end as u64;
             let num_rows = range.end - range.start;
-            let schema = schema.clone();
-            let mut decode_scheduler =
-                DecodeBatchScheduler::new(&schema, &column_infos, &Vec::new());
-
-            let (tx, rx) = mpsc::unbounded_channel();
-
-            decode_scheduler
-                .schedule_range(range.clone(), tx, &scheduler)
-                .await
-                .unwrap();
-
-            const BATCH_SIZE: u32 = 100;
-            let mut decode_stream = BatchDecodeStream::new(rx, BATCH_SIZE, num_rows).into_stream();
-
-            let mut offset = range.start as usize;
-            while let Some(batch) = decode_stream.next().await {
-                let batch = batch.await.unwrap().unwrap();
-                let actual = batch.column(0);
-                let expected = data.slice(offset, BATCH_SIZE as usize);
-                assert_eq!(expected.data_type(), actual.data_type());
-                assert_eq!(&expected, actual);
-                offset += BATCH_SIZE as usize;
-            }
+            let expected = data.slice(range.start as usize, num_rows as usize);
+            let scheduler = scheduler.clone();
+            test_decode(
+                num_rows,
+                &schema,
+                &column_infos,
+                expected,
+                |mut decode_scheduler, tx| {
+                    async move { decode_scheduler.schedule_range(range, tx, &scheduler).await }
+                        .boxed()
+                },
+            )
+            .await;
         }
+
+        // Test take scheduling
+        for indices in [
+            vec![100],
+            vec![0],
+            vec![9999],
+            vec![100, 1100, 5000],
+            vec![1000, 2000, 3000],
+            vec![2000, 2001, 2002, 2003, 2004],
+            // Big take that spans multiple pages and generates multiple output batches
+            (100..500).map(|i| i * 3).collect::<Vec<_>>(),
+        ] {
+            let num_rows = indices.len() as u64;
+            let indices_arr = UInt32Array::from(indices.clone());
+            let expected = arrow_select::take::take(&data, &indices_arr, None).unwrap();
+            let scheduler = scheduler.clone();
+            test_decode(
+                num_rows,
+                &schema,
+                &column_infos,
+                expected,
+                |mut decode_scheduler, tx| {
+                    async move {
+                        decode_scheduler
+                            .schedule_take(&indices, tx, &scheduler)
+                            .await
+                    }
+                    .boxed()
+                },
+            )
+            .await;
+        }
+
+        // Test large take that is bigger than batch size
```

Comment:
This a TODO?

Reply:
Yes, I will clean this up

Reply:
Actually, it *was* a TODO but I ended up adding it above so I just removed the comment.